app/src/main/java/com/banuba/sdk/example/beautification/MainActivity.java

```diff
@@ -176,44 +217,23 @@ protected void onStop() {
     @Override
     protected void onDestroy() {
         mCurrentEffect = null;
-        mSdkManager.releaseSurface();
+        if (mSdkManager != null) {
+            mSdkManager.releaseSurface();
+        }
         super.onDestroy();
     }
 
-    @Override
-    public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
-        String selected = (String) ((TextView) view).getText();
-        loadEffect(mEffects.get(selected));
+    private String makeJsMethod(String method, String params) {
+        return method + (params == null ? "()" : "(" + params + ")");
     }
 
     private void loadEffect(String effect) {
         mCurrentEffect = mSdkManager.loadEffect(BanubaSdkManager.getResourcesBase() + "/effects/" + effect, false);
         mEffectController.onEffectChanged(effect);
-    }
 
-    @Override
-    public void onNothingSelected(AdapterView<?> adapterView) {
-    }
-
-    @Override
-    public void onColorChanged(int argb) {
-        Log.d("color json", "ARGB color: " + String.format("%08X", argb));
-
-        // clang-format off
-        int a = (argb >> 24) & (0x000000FF);
-        int r = (argb >> 16) & (0x000000FF);
-        int g = (argb >> 8) & (0x000000FF);
-        int b = argb & (0x000000FF);
-        // clang-format on
-
-        List<Float> color_arr = new ArrayList<>(4);
-        color_arr.add((float) r / 0xFF);
-        color_arr.add((float) g / 0xFF);
-        color_arr.add((float) b / 0xFF);
-        color_arr.add((float) a / 0xFF);
-
-        String json = new JSONArray(color_arr).toString();
-        Log.d("color json", "json: " + json);
-        mCurrentEffect.callJsMethod("setColor", json);
+        /* fix: clear Hair color */
+        if (effect.equals("Makeup")) {
+            onSetterEvent("makeup.Hair.clear");
+        }
```

Comment:
instead of making this fix in code it is better to fix an effect - just drop some function call which leads to hair coloring in config.js